yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -189,6 +190,39 @@ private[spark] abstract class YarnSchedulerBackend(
   }
 
   /**
+   * Renew and update the credentials. This method will trigger credentials renewal in the AM
+   * side, once successfully renewed, it will trigger credentials updating in executor and
+   * driver side.
+   */
+  def updateCredentials(): Unit = {
+    yarnSchedulerEndpoint.amEndpoint match {
+      case Some(am) =>
+        val future = am.ask[Boolean](UpdateCredentials)
+
+        future.onSuccess {
+          case true =>
+            // Update credentials in the executor side only when AM is successfully renewed the
+            // Credentials.
+            synchronized {
+              executorDataMap.values.foreach(_.executorEndpoint.send(UpdateCredentials))
```

Comment:
Use ask instead of send.
This allows for consumers of the api to have reasonable confidence (based on Future's timeout) whether the update was propagated.

send could result in a task schedule before the update is processed.